Server/src/models/DatasetModels/DatasetApprovalModel.ts

```diff
@@ -0,0 +1,88 @@
+import { Connection, getConnection } from "typeorm";
+import { Dataset } from "../entities/Dataset";
+import { selectUnapprovedDatasetInfoQuery, Unapproveddatasets } from "../entities/Unapproveddatasets";
+import { IDatasetIDModel } from "../interfaces/DatasetModelInterface";
+import { DatasetCommonModel } from "./DatasetCommonModel";
+
+
+
+export class DatasetApprovalModel {
+    private connection: Connection;
+    private commonModel: DatasetCommonModel
+    constructor() {
+        this.connection = getConnection();
+        this.commonModel = new DatasetCommonModel();
+    }
+
+    private selectDatasetCommentQuery = (id: number) =>

```

Comment:
shouldnt this be async?

Reply:
also does it have return type?

Reply:
Will need to move of these methods around since the selects are general methods applicable to the entity and not only the model

Reply:
Queries do not need to be async and they also do not have a specific return type.